LDAR_Sim/src/virtual_world/sites.py

```diff
@@ -154,27 +164,107 @@ def _create_equipment_groups(
                 )
         elif isinstance(equipment_groups, (int, float)) and equipment_groups == 0:
             # special case for when equipment group isn't set
-            equip_count = math.ceil(
+            if (
                 propagating_params[Infrastructure_Constants.Sites_File_Constants.REP_EMIS_EPR]
-                * 365
-                * 2
-            )
-            equip_group_info = pd.Series({PLACEHOLDER_EQUIPMENT: equip_count})
+                is None
+                and propagating_params[
+                    Infrastructure_Constants.Sites_File_Constants.NON_REP_EMIS_EPR
+                ]
+                is None
+            ):
+                print(EMISSION_PRODUCTION_RATE_ERROR)
+                sys.exit()
+            elif (
+                propagating_params[Infrastructure_Constants.Sites_File_Constants.NON_REP_EMIS_EPR]
+                is None
+            ):
+                prod_rate = propagating_params[
+                    Infrastructure_Constants.Sites_File_Constants.REP_EMIS_EPR
+                ]
+                placeholder = PLACEHOLDER_REP_EQUIPMENT
+                print(
+                    PLACEHOLDER_CREATION_WARNING_MESSAGE.format(
+                        type="repairable", site=self._site_ID
+                    )
+                )
+            elif (
+                propagating_params[Infrastructure_Constants.Sites_File_Constants.REP_EMIS_EPR]
+                is None
+            ):
+                prod_rate = propagating_params[
+                    Infrastructure_Constants.Sites_File_Constants.NON_REP_EMIS_EPR
+                ]
+                placeholder = PLACEHOLDER_NON_EQUIPMENT
+                print(
+                    PLACEHOLDER_CREATION_WARNING_MESSAGE.format(
+                        type="non-repairable", site=self._site_ID
+                    )
+                )
+            else:
+                prod_rate = max(
+                    propagating_params[Infrastructure_Constants.Sites_File_Constants.REP_EMIS_EPR],
+                    propagating_params[
+                        Infrastructure_Constants.Sites_File_Constants.NON_REP_EMIS_EPR
+                    ],
+                )
+                placeholder = PLACEHOLDER_EQUIPMENT
+            equip_count = math.ceil(prod_rate * 365 * 2)
+            equip_group_info = pd.Series({placeholder: equip_count})
             prop_params = copy.deepcopy(propagating_params)
             self._equipment_groups.append(
                 Equipment_Group(
                     equipment_groups, infrastructure_inputs, prop_params, equip_group_info
```

Comment:
Make sure to divide up the site level production rate among the components somewhere. I recommend where the components are created.

Reply:
This will be addressed in a separate PR